### ISSUE TITLE
SSL handshake failed- Android issue

### ISSUE THREAD

**Issue by shah-pooja** [silkimen/cordova-plugin-advanced-http#150]
I am trying to login using the post method. The content type in the header is **"x-www-form-urlencoded"** . While doing login in the android devices it gives me error as following
**api response error** : {status: -1, error: "SSL handshake failed"}
**android studio error** : D/NativeCrypto: ssl=0x7933868700 NativeCrypto_SSL_do_handshake exception => 0
I/System.out: close [socket]
D/WebView: evaluateJavascript=cordova.callbackFromNative('CordovaHttpPlugin1822917423',false,9,[{"status":-1,"error":"SSL handshake failed"}],false);  
Using the same API I am trying to login in the ios devices it works fine.
Please help me out as soon as possible. Thank you.

**Comment by smartameer** [comment]
Experiencing Same issue.

**Comment by SaltyPotato** [comment]
Experiencing same issue.

**Comment by silkimen** [comment]
Hi guys, this is happening when something goes wrong during SSL handshake. Maybe the server is using a cipher suite or a SSL version which is not supported on the device? It's hard to tell without having more information. Maybe you can check the server logs to get more detailed infos?

**Comment by shah-pooja** [comment]
Upgrade the plugin to the latest version and before calling post method for api add the following code:
cordova.plugin.http.setSSLCertMode('nocheck', function() {
		console.log('success!');
       }, function() {
		console.log('error :(');
	});
The details about this method is already available in the plugin document. This helped me.

**Comment by Hanzofm** [comment]
Same problem with Android 6.0.1 what would be the problem with it?

**Comment by mirfanazam** [comment]
I am facing the same issue in Ionic app. I am calling this.https.setSSLCertMode('pinned'); before post request.

Can I have sample code and location (for android) to copy the certificates.

I have two certificate one .cer and other .p12. In browser i can make successful requests to the server but through device i get "SSL Handshake failed."

**Comment by danishin** [comment]
I think it's related to https://stackoverflow.com/a/36892715/3891342.
It happens for android < 5.0
Gonna fork and try to fix this issue!

**Comment by alemit** [comment]
> Experiencing same issue. Is this issue related to https://issuetracker.google.com/issues/37122132. I'm getting SSL handshake error on 7.0 devices, 7.1.1 and others work fine.

It is exactly this issue, but it is not resolved. Does someone resolved it somehow? Android 7.0 has ~10% market share and cannot be just skipped

**Comment by cvettelina** [comment]
It seems the proposed solution above
cordova.plugin.http.setSSLCertMode('nocheck', function() {
console.log('success!');
}, function() {
console.log('error :(');
});

works only once (for the first call) after that it again check the certs and the call fails for android 7.0 
Can anyone please advice when and how this function should be used in order to stop checking certificates

**Comment by shah-pooja** [comment]
please write in your service file where post method is defined. Hence every time you call post method this method will be executed along.

**Comment by Manu0086** [comment]
Same problem only on Android 7.0

**javax.net.ssl.sslhandshakeexception: handshake failed**

and _setSSLCertMode('nocheck');_ doesn't work !

**Comment by marutifh** [comment]
> I am trying to login using the post method. The content type in the header is **"x-www-form-urlencoded"** . While doing login in the android devices it gives me error as following
> **api response error** : {status: -1, error: "SSL handshake failed"}
> **android studio error** : D/NativeCrypto: ssl=0x7933868700 NativeCrypto_SSL_do_handshake exception => 0
> I/System.out: close [socket]
> D/WebView: evaluateJavascript=cordova.callbackFromNative('CordovaHttpPlugin1822917423',false,9,[{"status":-1,"error":"SSL handshake failed"}],false);
> Using the same API I am trying to login in the ios devices it works fine.
> Please help me out as soon as possible. Thank you.

Did you find any solution? I am facing the same issue, and this is not with any Android version specific, I am facing this issue with all Android versions.

**Comment by ommyjay** [comment]
Try using `acceptAllCerts(true)` method before the request

**Comment by silkimen** [comment]
Closing this, feel free to open a new issue, if this won't solve your problem! 👍
